features/step_definitions/helper_services.rb

```diff
@@ -153,27 +153,28 @@
   step %Q/evaluation of `pod` is stored in the :proxy_pod clipboard/
 end
 
-Given /^I have LDAP service in my project$/ do
-    ###
-    # The original idea is trying to put ldap server in the openshift to make this flexy.
-    # Since we run the scenario in jenkins slave which is not in sdn, then two choices come to me:
-    # 1, Create a route for the ldapserver pod, but blocked by this us https://trello.com/c/9TXvMeS2 is done.
-    # 2, Port forward the ldap server pod to the jenkins slave.
-    # So take the second one since this one can be implemented currently
-    ###
+###
+# The original idea is trying to put ldap server in the openshift to make this flexy.
+# Since we run the scenario in jenkins slave which is not in sdn, then two choices come to me:
+# 1, Create a route for the ldapserver pod, but blocked by this us https://trello.com/c/9TXvMeS2 is done.
+# 2, Port forward the ldap server pod to the jenkins slave.
+# So take the second one since this one can be implemented currently
+###
+Given /^I have LDAP service in my project$/ do 
     step %Q/I run the :run client command with:/, table(%{
       | name  |ldapserver                             |
       | image |openshift/openldap-2441-centos7:latest |
       })
     step %Q/the step should succeed/
-    step %Q/I wait until replicationController "ldapserver-1" is ready/
-
+    step %Q/a pod becomes ready with labels:/, table(%{
+      | run=ldapserver |
+      })
+    
     cb.ldap_pod = BushSlicer::Pod.get_labeled(["run", "ldapserver"], user: user, project: project).first
```

Comment:
`cb.ldap_pod = pod`

Reply:
the step before will ache the pod as current one, that's why `pod` will return the one we care about.

Reply:
testing this, once complete I will push this changes to PR

Reply:
This step is required for lower version than 4.5 if I removed this line, its failing.  If I use `pod.name` it works for > 4.5 version. As its taking pod.name as `ldapserver` but in < 4.5 version pod are created as like `ldapserver-1-gzdld` and step is failed as its searching for `ldapserver` . Basically `pod.name` should take the name which is created but for some reason its always taking `ldapserver`

Reply:
What I tried to say is that the step above `a pod becomes ready with labels` already sets current pod to a ready pod with label `run=ldapserver`. So you don't need to do that again in this line. This line can be removed and replaced by `cb.ldap_pod = pod`.